src/util/WalletBackendAdapter.ts

```diff
@@ -188,15 +205,79 @@ export const makeWalletBridgeFromFollowers = (
     marshaller,
   );
 
+  const { notifier: beansOwingNotifier, updater: beansOwingUpdater } =
+    makeNotifierKit<Number | null>(null);
+
   const watchBeansOwing = async () => {
     for await (const { value } of iterateLatest(beansOwingFollower)) {
+      if (isHalted) return;
       beansOwingUpdater.updateState(Number(value));
     }
   };
 
+  // Reads purses from the cosmos bank module. These are not necessarily real
+  // "purses" in the smart wallet contract, because those are lazily
+  // instantiated, but we still need to show the balances.
+  const watchChainBalances = () => {
+    let vbankAssets;
+    let bank;
+
+    const possiblyUpdateBankPurses = () => {
+      if (!vbankAssets || !bank) return;
+
+      const bankMap = new Map<string, string>(
+        bank.map(({ denom, amount }) => [denom, amount]),
+      );
+
+      vbankAssets.forEach(([denom, info]) => {
+        const amount = bankMap.get(denom) ?? 0n;
```

Comment:
If an asset exists in the vbank we want to show it as a purse with 0 balance so they can click the eventual "deposit" button for ibc transfer.

Reply:
Very helpful note. Please include in the code.

Reply:
Done